File: day_three/python/arithemetic_challenge.py
import random

def generate_arithmetic_problem():
	first_number = random.randint(1, 10)
	second_number = random.randint(1, 10)
	operator = ("*", "+", "-", "/")
	selected_operator = random.choice(operator)
	calculation = f"{first_number} {selected_operator} {second_number}"
	if selected_operator == "+":
        	correct_answer = first_number + second_number
	elif selected_operator == "-":
        	correct_answer = first_number - second_number
	elif selected_operator == "*":
        	correct_answer = first_number * second_number
	else:
        	correct_answer = first_number / second_number 
	return selected_operator, calculation, correct_answer






total_attempts = 10
correct_attempts = 0

print("Welcome to the Arithmetic Challenge!\nYou have 10 attempts to solve random arithmetic problems.")
print()
for attempt in range(total_attempts):
	correct_operator, problem, correct_answer = generate_arithmetic_problem()
	user_answer = float(input(f"Attempt {attempt + 1}: What is {problem} ? "))

	if user_answer == correct_answer:
		print("Correct!  ")
		correct_attempts += 1
	else:
		print(f"Incorrect. The correct answer was {correct_answer}.")

print("\nChallenge completed!")
print(f"Total correct attempts: {correct_attempts}/{total_attempts}")
print(f"Your final score: {correct_attempts * 10}%")

File: day_three/javascripts/arithemeticChallenge.js
const prompt = require("prompt-sync")();
function generateArithmeticProblem() {
    const firstNumber = Math.floor(Math.random() * 10) + 1;
    const secondNumber = Math.floor(Math.random() * 10) + 1;
    const operators = ["*", "+", "-", "/"];
    const selectedOperator = operators[Math.floor(Math.random() * operators.length)];
    const calculation = `${firstNumber} ${selectedOperator} ${secondNumber}`;

    let correctAnswer;
    switch (selectedOperator) {
        case "+":
            correctAnswer = firstNumber + secondNumber;
            break;
        case "-":
            correctAnswer = firstNumber - secondNumber;
            break;
        case "*":
            correctAnswer = firstNumber * secondNumber;
            break;
        case "/":
            correctAnswer = firstNumber / secondNumber;
            break;
    }

    return { selectedOperator, calculation, correctAnswer };
}

const totalAttempts = 10;
let correctAttempts = 0;

console.log("Welcome to the Arithmetic Challenge!\nYou have 10 attempts to solve random arithmetic problems.\n");

for (let attempt = 1; attempt <= totalAttempts; attempt++) {
    const { selectedOperator, calculation, correctAnswer } = generateArithmeticProblem();
    const userAnswer = parseFloat(prompt(`Attempt ${attempt}: What is ${calculation}? `));

    if (userAnswer === correctAnswer) {
        console.log("Correct!\n");
        correctAttempts++;
    } else {
        console.log(`Incorrect. The correct answer was ${correctAnswer}.\n`);
    }
}

console.log("Challenge completed!");
console.log(`Total correct attempts: ${correctAttempts}/${totalAttempts}`);
console.log(`Your final score: ${correctAttempts * 10}%`);
